akka-stream/src/main/scala/akka/stream/Attributes.scala

```diff
@@ -311,6 +312,85 @@ object Attributes {
       extends Attribute
   final case object AsyncBoundary extends Attribute
 
+  /**
+   * Cancellation strategies provide a way to configure the behavior of a stage when `cancelStage` is called.
+   *
+   * It is only relevant for stream components that have more than one output and do not define a custom cancellation
+   * behavior by overriding `onDownstreamFinish`. In those cases, if the first output is cancelled, the default behavior
+   * is to call `cancelStage` which shuts down the stage completely. The given strategy will allow customization of how
+   * the shutdown procedure should be done precisely.
+   */
+  @ApiMayChange
+  final case class CancellationStrategy(strategy: CancellationStrategy.Strategy) extends Attribute
+  @ApiMayChange
+  object CancellationStrategy {
+    val Default: CancellationStrategy = CancellationStrategy(PropagateFailure)
```

Comment:
I think it would be better as a `MandatoryAttribute` and then always put the default in the attributes and use `mandatoryAttribute` to access it rather than each use sites saying "use this or a default that I provide". Did you see any problems with that?

Reply:
That's probably right. The biggest problem I have is still with adding new settings to `ActorMaterializerSettings` with confidence.

Reply:
I'm considering now to keep it just like that for now and fix the configuration of the default in a separate PR to unblock this one.

Reply:
I guess that can still be done completely internal so post RC1 will be ok.

Reply:
Changed it to a mandatory attribute that is set in `ActorMaterializerSettings.toAttributes` from the default defined here.

Reply:
No, let's make it `private[stream]`.

Reply:
Filed https://github.com/akka/akka/issues/28000 to make it configurable later on.